tick/optim/model/huber.py

```diff
@@ -0,0 +1,127 @@
+        \\end{cases}
+
+    for :math:`y, y' \\in \\mathbb R`, where :math:`\\delta > 0` can be tuned
+    using the ``threshold`` argument. Data is passed to this model through the
+    ``fit(X, y)`` method where X is the features matrix (dense or sparse) and
+    y is the vector of labels.
+
+    Parameters
+    ----------
+    fit_intercept : `bool`
+        If `True`, the model uses an intercept
+
+    Attributes
+    ----------
+    features : {`numpy.ndarray`, `scipy.sparse.csr_matrix`}, shape=(n_samples, n_features)
+        The features matrix, either dense or sparse
+
+    labels : `numpy.ndarray`, shape=(n_samples,) (read-only)
+        The labels vector
+
+    n_samples : `int` (read-only)
+        Number of samples
+
+    n_features : `int` (read-only)
+        Number of features
+
+    n_coeffs : `int` (read-only)
+        Total number of coefficients of the model
+
+    n_threads : `int`, default=1 (read-only)
+        Number of threads used for parallel computation.
+
+        * if ``int <= 0``: the number of physical cores available on
+          the CPU
+        * otherwise the desired number of threads
+    """
+
+    _attrinfos = {
+        "threshold": {
+            "writable": True,
+            "cpp_setter": "set_threshold"
+        }
+    }
+
+    def __init__(self, fit_intercept: bool = True, threshold: float = 1,
+                 n_threads: int = 1):
+        ModelFirstOrder.__init__(self)
+        ModelGeneralizedLinear.__init__(self, fit_intercept)
+        ModelLipschitz.__init__(self)
+        self.n_threads = n_threads
+        self.threshold = threshold
+
+    # TODO: implement _set_data and not fit
+    def fit(self, features, labels):
+        """Set the data into the model object
+
+        Parameters
+        ----------
+        features : {`numpy.ndarray`, `scipy.sparse.csr_matrix`}, shape=(n_samples, n_features)
+            The features matrix, either dense or sparse
+
+        labels : `numpy.ndarray`, shape=(n_samples,)
+            The labels vector
+
+        Returns
+        -------
+        output : `ModelHuber`
+            The current instance with given data
+        """
+        ModelFirstOrder.fit(self, features, labels)
+        ModelGeneralizedLinear.fit(self, features, labels)
+        ModelLipschitz.fit(self, features, labels)
+        self._set("_model", _ModelHuber(self.features,
+                                        self.labels,
+                                        self.fit_intercept,
+                                        self.threshold,
+                                        self.n_threads))
+        return self
+
+    def _grad(self, coeffs: np.ndarray, out: np.ndarray) -> None:
+        self._model.grad(coeffs, out)
+
+    def _loss(self, coeffs: np.ndarray) -> float:
+        return self._model.loss(coeffs)
+
+    def _get_lip_best(self):
+        # TODO: Use sklearn.decomposition.TruncatedSVD instead?
```

Comment:
Remaining TODO. What would be the advantages of doing so? Imo, relying on sklearn outside the inference submodule is not great, but I don't know what you decided on this matter. @Mbompr ?

Reply:
Same as before, to be decided in another PR. Let's keep it like this for now (was already there before for other models...)